docs/src/main/sphinx/routines/function.md

```diff
@@ -0,0 +1,92 @@
+# FUNCTION
+
+## Synopsis
+
+```text
+FUNCTION name [ (parameter1, parameter2, ...) ]
+  RETURNS type
+  [ LANGUAGE langaugeIdentifier]
+  [ NOT? DETERMINISTIC ]
+  [ RETURNS NULL ON NULL INPUT ]
+  [ CALLED ON NULL INPUT ]
+  [ SECURITY { DEFINER | INVOKER } ]
+  [ COMMENT description]
+  statements
+```
+
+## Description
+
+Declare a SQL routine.
+
+The `name` must qualify the name of the catalog to store the routine, unless the
+function is only declared in the scope of a query.
+
+The optional list of parameters is a comma-separated list of names and
+data types.
+
+The `type` value after the `RETURNS` keyword identifies the [data type]() of the
+routine output.
+
+The optional `LANGUAGE` statement identifies the language used for the routine
+definition with `languageIdentifier`. Only `SQL` is supported.
+
+The optional `DETERMINISTIC` or `NOT DETERMINISTIC` statement declared to the
+query engine, if repeated functions calls with identical input parameters yield
+the same result.
+
+The optional `RETURNS NULL ON NULL INPUT` is used to declare that the function
+returns a `NULL` value when `NULL` values are provided as input. This
+declaration can be used by the engine to avoid processing with `NULL`` input since
+the result is known and declared preemptively.
+
+The `CALLED ON NULL INPUT` explicitly declares to the engine that the routine
+must be processed with `NULL` input parameter values.
+
+The security declaration of `SECURITY INVOKER` or `SECURITY DEFINER` is only
+valid for catalog routines. It sets the mode for processing the routine with the
+permissions of the user who calls the routine (`INVOKER`) or who created the
+routine in the catalog (`DEFINER`).
+
+On optional `description` string after the keyword `COMMENT` can be used to
+provide information about the function to other users. The information is
+accessible with [](/sql/show-functions).
+
+The main definition of the routine is defined as `statements` by a simple
+`RETURN` statement with an expression, or a compound list of statements in a
+`BEGIN` block.
+
+## Examples
+
+A simple catalog function:
+
+```sql
+CREATE FUNCTION example.meaning_of_life()
```

Comment:
Missing schema. Same for all of the below.

Reply:
Done